lua/lazyvim/plugins/extras/lang/go.lua

```diff
@@ -15,6 +15,10 @@ return {
     opts = {
       servers = {
         gopls = {
+          keys = {
+            -- Workaround for the lack of a DAP strategy in neotest-go: https://github.com/nvim-neotest/neotest-go/issues/12
+            { "<leader>td", "<cmd>lua require('dap-go').debug_test()<CR>", desc = "Debug Nearest (Go)" },
```

Comment:
Are we sure this will always overwrite the neotest configuration?
I would suspect that it does not, if you load the go lang extra before the neotest extra?

Also, this will unconditionally bind `<leader>td` even if we do not have the neotest or dap extras.
I think this keybind probably has to be done under an optional neotest dependency, probably under the `nvim-dap` dependency too 😄

Reply:
I'm kind of new to Neovim/LazyVim and can't say I know for sure what's the best way to do this, or what caveats/side-effects there may be. Would be great to have @folke pitch in here (but I see he's on vacation so it can wait until he's back).

But since I think you had an interesting observation, I just now tried to load the go extras _before_ the neotest/dap extras, just to see if the keymap would still be overridden for go tests. I initially did this:

```lua
--  lazy.lua
...

require("lazy").setup({
  spec = {
    -- add LazyVim and import its plugins
    { "LazyVim/LazyVim", import = "lazyvim.plugins" },

    -- import any extras modules here
    { import = "lazyvim.plugins.extras.lang.go" },
    { import = "lazyvim.plugins.extras.test.core" },
    { import = "lazyvim.plugins.extras.dap.core" },

    { import = "plugins" },
  },

...
```

I also tried switching them around, and in _all_ cases, I were able to debug a go test with `<leader>td` - which means the keymap seems like it's always overridden regardless of this order.

However, I'm not fully understood with how the Lazy.nvim `dependencies =` works, and perhaps my local config somehow triggers loading of plugins in some order that doesn't show by this rudimental test of mine.

Reply:
yes, should be fine